### PR TITLE
Docs: Provide descriptive labels for http references

### LINKED ISSUE #78899: Docs: Replace occurrences of "See http://" with a descriptive label in 4 files
### Summary

Accessibility guidelines recommend we do not use "See http://<website>" in documentation, but instead provide context around this for screen readers etc.

In this issue, we've identified 4 files that use this convention in the documentation. For each occurrence, replace it with an RST link to [external web page](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/references.html#links-to-external-web-pages) .

Specifically, use the format \`descriptive phrase \<url\>\`_

List of affected RST pages are in a follow-on comment. You can choose to fix one at a time, using the Edit on GitHub link at the top of the RST page, or in one PR to fix them both.

### Issue Type

Documentation Report

### Component Name

rst/dev_guide/testing/sanity/mypy.rst

### Ansible Version

```console
$ ansible --version
2.14
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
none
```

### OS / Environment

none

### Additional Information

none

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by Blaster4385**
##### SUMMARY

- This PR replaces multiple occurrences of "See http://" with corresponding descriptive labels.
- Fixes #78899 

##### ISSUE TYPE

- Docs Pull Request

### CODE REVIEW

**Review by CaedenPH — APPROVED** [review]

**Review by oraNod — APPROVED** [review]

File: docs/docsite/rst/playbook_guide/playbooks_filters.rst [hunk 1]
```diff
@@ -2106,7 +2106,7 @@ To get a date object from a string use the `to_datetime` filter:
     # get amount of days between two dates. This returns only number of days and discards remaining hours, minutes, and seconds
     {{ (("2016-08-14 20:00:12" | to_datetime) - ("2015-12-25" | to_datetime('%Y-%m-%d'))).days  }}
 
-.. note:: For a full list of format codes for working with python date format strings, see https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior.
+.. note:: For a full list of format codes for working with python date format strings, see the `python datetime documentation <https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior>`.
```
Comment (samccann) [inline docs/docsite/rst/playbook_guide/playbooks_filters.rst]:
```suggestion
.. note:: For a full list of format codes for working with python date format strings, see the :ref:`python3.datetime`.
```

**Comment by samccann** [comment]
Hi @Blaster4385 and thanks for the hacktoberfest docs fix for Ansible!  I pushed one minor change that I want to test out. To explain - we use the [intersphinx](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html) to shorten links to other projects that support this. Python is one of these projects so I replaced the hard URL with a sphinx/rst :ref:. I'm testing locally to be sure I did the right thing.  The benefit of this over a hard URL is that if the Python docs team changes the location of this information, the intersphinx link will remain valid, where a hard URL may not.

**Comment by samccann** [comment]
...and my test failed. Sorry about that. I'll replace your fix and merge this asap.

File: docs/docsite/rst/playbook_guide/playbooks_filters.rst [hunk 2]
```diff
@@ -2106,7 +2106,7 @@ To get a date object from a string use the `to_datetime` filter:
     # get amount of days between two dates. This returns only number of days and discards remaining hours, minutes, and seconds
     {{ (("2016-08-14 20:00:12" | to_datetime) - ("2015-12-25" | to_datetime('%Y-%m-%d'))).days  }}
 
-.. note:: For a full list of format codes for working with python date format strings, see https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior.
+.. note:: For a full list of format codes for working with python date format strings, see the :ref:`python3.datetime`.
```
Comment (samccann) [inline docs/docsite/rst/playbook_guide/playbooks_filters.rst]:
```suggestion
.. note:: For a full list of format codes for working with python date format strings, see the `python datetime documentation <https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior>`_.
```

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
Thanks @Blaster4385 for the hacktoberfest Ansible docs fix and welcome to the Ansible project!

**Comment by Blaster4385** [comment]
> Thanks @Blaster4385 for the hacktoberfest Ansible docs fix and welcome to the Ansible project!

Thanks for accepting my contribution. As for that intersphinx link, I had already tested that and it was failing tests, hence, I changed it to a hard URL. Anyways, thanks for such a warm welcome and I hope to contribute more to the project.

**Pull request merged**